graph-builder/src/registry.rs

```diff
@@ -37,23 +45,109 @@ impl Into<cincinnati::Release> for Release {
     }
 }
 
+type RegistryClient = dkregistry::v2::Client;
+
+fn registry_trim(src: &str) -> &str {
+    src.trim_left_matches("https://")
+        .trim_left_matches("http://")
+}
+
+fn get_credentials_or_empty_reader(
+    credentials_file: &str,
+) -> Result<Box<dyn std::io::Read>, Error> {
+    if credentials_file != "" {
+        match File::open(credentials_file) {
+            Ok(file) => Ok(Box::new(file)),
+            Err(err) => bail!(err),
+        }
+    } else {
+        Ok(Box::new(std::io::empty()))
+    }
+}
+
+fn get_authenticated_client(
+    tcore: &mut Core,
+    host: &str,
+    image: &str,
+    credentials_file: &str,
+) -> Result<RegistryClient, Error> {
+    let mut registry_client = dkregistry::v2::Client::configure(&tcore.handle())
+        .registry(host)
+        .insecure_registry(false)
+        .read_credentials(get_credentials_or_empty_reader(credentials_file)?)
+        .build()
+        .map_err(|e| format_err!("{}", e))?;
```

Comment:
This shouldn't be necessary.

Reply:
Unfortunately there's no obvious better way to solve https://github.com/rust-lang-nursery/error-chain/issues/240.